internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -53,6 +53,9 @@ func (k *Kubelet) Condition(data *userdata.UserData) conditions.Condition {
 
 // DependsOn implements the Service interface.
 func (k *Kubelet) DependsOn(data *userdata.UserData) []string {
+	if data.Services.Kubeadm.IsControlPlane() {
+		return []string{"containerd", "etcd", "kubeadm"}
```

Comment:
I wonder if we should depend on etcd here. Do we gain anything? Probably faster bootstrap times.

Reply:
Nope. Kubeadm still depends on etcd. Maybe when we move to bootkube we will get this benefit.